### PR TITLE
Gracefully handle format exception from Diagnostic.GetMessage() when …

### LINKED ISSUE #4589: DateTime SystemFormat Exception on Workspaces.dll
Managed Debugging Assistant 'FatalExecutionEngineError' has detected a problem in 'C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\devenv.exe'.

Additional information: The runtime has encountered a fatal error. The address of the error was at 0x228a06cf, on thread 0xd0c. The error code is 0x80131623. This error may be a bug in the CLR or in the unsafe or non-verifiable portions of user code. Common sources of this bug include user marshaling errors for COM-interop or PInvoke, which may corrupt the stack

http://www.mediafire.com/view/a8x4vpyyamv8wtp/buildlog.txt
http://www.mediafire.com/view/ax9hdo4d9a16m60/debuglog.txt

Tested on a newly installed Visual Studio 2015 Community Edition in both Windows 8.1 and 10

Edit: Error occurs when opening a project created in earlier Visual Studio versions; works on projects created in VS 2015

### PR DESCRIPTION
**Pull request by mavasani**
…the analyzer reports a diagnostic with incorrect format arguments.

Fixes #4589

### CODE REVIEW

**Comment by mavasani** [comment]
ping.. this is a pretty trivial fix.

File: src/Compilers/Core/Portable/Diagnostic/Diagnostic_SimpleDiagnostic.cs [hunk 1]
```diff
@@ -99,7 +99,16 @@ public override string GetMessage(IFormatProvider formatProvider = null)
...
-                return string.Format(formatProvider, localizedMessageFormat, _messageArgs);
+
+                try
+                {
+                    return string.Format(formatProvider, localizedMessageFormat, _messageArgs);
+                }
+                catch (Exception)
+                {
+                    // Analyzer reported diagnostic with invalid format arguments, so just return the unformatted message.
+                    return localizedMessageFormat;
```
Comment (JohnHamby) [inline src/Compilers/Core/Portable/Diagnostic/Diagnostic_SimpleDiagnostic.cs]: Would there be any value in reporting that the formatting failed?

**Comment by JohnHamby** [comment]
I sign off.

File: src/Compilers/Core/Portable/Diagnostic/Diagnostic_SimpleDiagnostic.cs (hunk 1 above)
Comment (mavasani) [inline src/Compilers/Core/Portable/Diagnostic/Diagnostic_SimpleDiagnostic.cs]: Probably, but that needs a bit of additional work as GetMessage is not invoked by the analyzer driver, but by the analyzer host (command line compiler, error list and IDE diagnostic service, etc.) and so each host would need to setup a handler for the exception and take action. I'd let it be as is until we get any customer request for it.

**Pull request merged**